src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -199,18 +209,17 @@ fn compile<'a, 'cfg>(targets: &[(&'a Target, &'a Profile)],
             });
 
             // Figure out what stage this work will go into
-            let dst = match (target.is_lib(),
+            let stage = match (target.is_lib(),
                              profile.test,
                              target.is_custom_build()) {
-                (_, _, true) => jobs.queue(pkg, Stage::BuildCustomBuild),
-                (true, true, _) => jobs.queue(pkg, Stage::LibraryTests),
-                (false, true, _) => jobs.queue(pkg, Stage::BinaryTests),
-                (true, false, _) => jobs.queue(pkg, Stage::Libraries),
-                (false, false, _) if !target.is_bin() => {
-                    jobs.queue(pkg, Stage::BinaryTests)
-                }
-                (false, false, _) => jobs.queue(pkg, Stage::Binaries),
+                (_, _, true) => Stage::BuildCustomBuild,
+                (true, true, _) => Stage::LibraryTests,
+                (false, true, _) => Stage::BinaryTests,
+                (true, false, _) => Stage::Libraries,
+                (false, false, _) if !target.is_bin() => Stage::BinaryTests,
+                (false, false, _) => Stage::Binaries,
             };
+            let dst = jobs.queue(pkg, stage);
```

Comment:
I don't think any logic changed here, right? Could this stay the same as it was before?